### ISSUE TITLE
[BUG] nullary constructor warning for c++ map const_iterator

### ISSUE THREAD

**Issue by fried** [cython/cython#4657]
**Describe the bug**
Changes to libcpp/map.pxd broke existing code to use const_find on a map.
I think it was broken by #4410 

```cython
...
    def __getitem__(self, str key):
        err = KeyError(key)
        if not self or key is None:
            raise err
        cdef string ckey = key.encode('utf-8')
        it = deref(self._getMap()).const_find(ckey)
       ^
------------------------------------------------------------

thrift.py3/common.pyx:47:8: C++ class must have a nullary constructor to be stack allocated
```

As a workaround: 
For our internal build of 3.0.0a10 i added nullary constructors to all those iterator types where an alternate constructor from the non const iterator was defined. 

**To Reproduce**
Code to reproduce the behaviour:
```cython
from libcpp.map import map
from libcpp.string import string

def test():
    cdef map[string, string] mymap
    cdef string ckey = b'somekey'
    it = mymap.const_find(ckley)
```

**Expected behavior**
It compiles like it used too. 

**Environment (please complete the following information):**
 - OS: Linux 
 - Python version: 3.8.6
 - Cython version: 3.0.0a10

**Additional context**
Broken in master afaict, from my inspection of the pxd

**Comment by da-woods** [comment]
Thanks for the report

> For our internal build of 3.0.0a10 i added nullary constructors to all those iterator types where an alternate constructor from the non const iterator was defined.

This is almost certainly the right fix, so a PR containing those changes would be very welcome

**Comment by maxbachmann** [comment]
@scoder @fried this was fixed as part of #4528
